include/obstacle_detector/utilities/segment.h

```diff
@@ -107,6 +107,15 @@ class Segment
     return (p - projection).length();
   }
 
+  bool isBetweenEndpoints(const Point& p, double tol) const {
+    // Checks if point is between segment endpings (within a tolerance to account for noise)
+    Point a = last_point - first_point;
+    Point b = p - first_point;
+    Point c = p - last_point;
+    double t = a.dot(b) / a.lengthSquared();
```

Comment:
Please don't. Don't use the "neighboring" code to affect your "engineering" style. I have worked with a bunch of scientific codes that used "p, q, r" etc as variable names because "it was clear from context what they meant"... Trust me it was not until I opened an algorithm book from which the code was implemented. Do not rely on comments to explain your meaning, as the comments will become stale or get ignored. We have a comment in our codebase that says "DO NOT REMOVE THE LINE BELOW OR SCANNER WILL BREAK" and someone still went and removed the line in question.